src/PrestaShopBundle/Twig/Component/Header.php

```diff
@@ -71,4 +64,77 @@ public function getJsDef(): array
     {
         return Media::getJsDef();
     }
+
+    public function getPsVersion(): string
+    {
+        return $this->psVersion;
+    }
+
+    public function getIsoUser(): string
+    {
+        return $this->context->getLanguage()->getIsoCode();
+    }
+
+    public function getCountryIsoCode(): string
+    {
+        return $this->context->getContext()->country->iso_code;
+    }
+
+    public function getLangIsRtl(): bool
+    {
+        return (bool) $this->context->getLanguage()->isRTL();
+    }
+
+    public function getShopName(): string
+    {
+        return $this->configuration->get('PS_SHOP_NAME');
+    }
+
+    public function getControllerName(): string
+    {
+        return htmlentities(Tools::getValue('controller'));
```

Comment:
I don't doubt legacy page handled this this but I'm really not sure it's a stable way to do this Especially if you use the legacy `Tools::getValue` method We try and build a clean new layout so we should not rely on legacy code as much as possible

The first way to get this controller name is to look into the request attributes, most of them have a `_legacy_controller` attribute that you can retrieve It's actually that way we fetch it in the legacy layout currently https://github.com/PrestaShop/PrestaShop/blob/8372f8390a3e259237a289e343dfff951ff25789/src/PrestaShopBundle/Resources/views/Admin/layout.html.twig#L27

Reply:
When the attribute is not present we need to think about potential fallback way to get it, but as you detected in the tests some pages simply don't specify this anywhere and we can't deduce it so I guess even if it's rare we must anticipate and handle pages that have no related controllers And this method can return an empty string (I'm not sure at all what side effects this can cause but I don't know what alternative we have for now)